Sources/ConsoleKit/Utilities/LoggerFragment.swift

```diff
@@ -0,0 +1,329 @@
+		output += self.literal
+	}
+}
+
+public struct SeparatorFragment<T: LoggerFragment>: LoggerFragment {
+	public let literal: ConsoleText
+	public var fragment: T
+	
+	public init(_ literal: ConsoleText, fragment: T) {
+		self.literal = literal
+		self.fragment = fragment
+	}
+	
+	public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+		if output.needsSeparator {
+			if self.fragment.hasContent(record: &record) {
+				output.needsSeparator = false
+				output += self.literal
+			}
+		}
+		
+		self.fragment.write(&record, to: &output)
+	}
+}
+
+/// Writes the logged message to the output, and requests a separator for the next fragment.
+public struct MessageFragment: LoggerFragment {
+	public init() { }
+	
+	public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+		output += record.message.description.consoleText()
+		output.needsSeparator = true
+	}
+}
+
+/// Writes the combined metadata to the output, and requests a separator for the next fragment only if the metadata was not empty.
+///
+/// This fragment is considered to not have content if the metadata is empty.
+public struct MetadataFragment: LoggerFragment {
+	public init() { }
+	
+	public func hasContent(record: inout LogRecord) -> Bool {
+		 !record.allMetadata().isEmpty
+	}
+	
+	public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+		let allMetadata = record.allMetadata()
+		
+		guard !allMetadata.isEmpty else { return }
+		
+		output += allMetadata.sortedDescriptionWithoutQuotes.consoleText()
+		output.needsSeparator = true
+	}
+}
+
+/// Writes the file location of the logged message, including the line. This fragment requests a separator for the next fragment.
+public struct SourceLocationFragment: LoggerFragment {
+	public init() { }
+	
+	public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+		let file = record.file + ":" + record.line.description
+		output += "(" + file.consoleText() + ")"
+		output.needsSeparator = true
+	}
+}
+
+public struct TimestampFragment: LoggerFragment {
+	public init() { }
+	
+	public func write(_ record: inout LogRecord, to output: inout FragmentOutput) {
+		if #available(macOS 11, iOS 14, tvOS 14, watchOS 7, *) {
+			output += self.timestampUninit().consoleText()
+		} else {
+			output += self.timestamp().consoleText()
+		}
+		
+		output.needsSeparator = true
+	}
+	
+	@available(macOS 11, iOS 14, tvOS 14, watchOS 7, *)
+	private func timestampUninit() -> String {
+		String(unsafeUninitializedCapacity: 255) {
+			#if canImport(CRT)
+				var timestamp = __time64_t()
+				var localTime = tm()
+				_ = _time64(&timestamp)
+				_ = _localtime64_s(&localTime, &timestamp)
+			#else
+				var timestamp = time(nil)
+				var localTime = tm()
+				localtime_r(&timestamp, &localTime)
+			#endif
+			return $0.withMemoryRebound(to: Int8.self) { strftime($0.baseAddress!, $0.count, "%Y-%m-%dT%H:%M:%S%z", &localTime) }
+		}
+	}
+	
```

Comment:
There's no need for providing the alternative approach; just always use the 10.15-compatible version.
```suggestion
        output += self.timestamp().consoleText()
        output.needsSeparator = true
    }	
```
(Also, please set your text editor to use spaces instead of tabs for indentation 😅)